src/main/java/seedu/address/model/task/Task.java

```diff
@@ -132,13 +132,14 @@ public boolean equals(Object other) {
         return otherTask.getName().equals(getName())
                 && otherTask.getTags().equals(getTags())
                 && otherTask.getDate().equals(getDate())
-                && otherTask.getDescription().equals(getDescription());
+                && otherTask.getDescription().equals(getDescription())
+                && otherTask.getPriority().equals(getPriority());
```

Comment:
if duplicate tasks are not detected as duplicate after edited could it be because the equals method now captures priority? so if priority != same then it counts as not duplicated?